File: SOURCE_EXTERNAL/VecompSoftware.DocSuite.ExternalViewer/src/app/models/fascicles/fascicle-folder.model.ts
import { BaseServiceModel } from '../base-service.model';

export class FascicleFolderModel implements BaseServiceModel {
    readonly serviceModelName: string;

    constructor() {
        this.serviceModelName = "FascicleFolderModel";
    }

    id: string;
    status: string;
    typology: string;
    name: string;
    idFascicle: string;
    idCategory: string;
    hasDocuments: boolean;
    hasChildren: boolean;
    fascicleFolderLevel: number;
}

File: SOURCE_WEBAPI/VecompSoftware.DocSuiteWeb.Model/Entities/Fascicles/FascicleFolderModel.cs
using System;
using System.Collections.Generic;
using VecompSoftware.DocSuiteWeb.Model.Entities.Commons;

namespace VecompSoftware.DocSuiteWeb.Model.Entities.Fascicles
{
    public class FascicleFolderModel
    {
        public Guid UniqueId { get; set; }

        public string Name { get; set; }

        public FascicleFolderStatus Status { get; set; }

        public FascicleFolderTypology Typology { get; set; }

        public Guid? IdFascicle { get; set; }

        public short? IdCategory { get; set; }

        public bool HasDocuments { get; set; }

        public bool HasChildren { get; set; }

        public short FascicleFolderLevel { get; set; }
        public DocumentModel Document { get; set; }
        public ICollection<FascicleDocumentModel> FascicleDocuments { get; set; }
    }
}
